lib/Minz/Session.php

```diff
@@ -4,18 +4,31 @@
  * La classe Session gère la session utilisateur
  */
 class Minz_Session {
+	private static $volatile = false;
+	
 	/**
 	 * Initialise la session, avec un nom
 	 * Le nom de session est utilisé comme nom pour les cookies et les URLs(i.e. PHPSESSID).
-	 * Il ne doit contenir que des caractères alphanumériques ; il doit être court et descriptif
+	 * Il ne doit contenir que des caractères alphanumériques ; il doit être court et descriptif.
+	 * If the volatile parameter is true, then no cookie and not session storage are used.
+	 * Volatile is especially useful for API calls without cookie / Web session.
 	 */
-	public static function init($name) {
+	public static function init($name, $volatile = false) {
+		self::$volatile = $volatile;
+		if (self::$volatile) {
+			$_SESSION = [];
+			return;
+		}
+
 		$cookie = session_get_cookie_params();
 		self::keepCookie($cookie['lifetime']);
 
 		// démarre la session
 		session_name($name);
 		session_start();
+		session_write_close();
+		//Use cookie only the first time the session is started
+		ini_set('session.use_cookies', '0');
```

Comment:
Needed, otherwise PHP will attempt to re-send some HTTP headers

Reply:
I don't understand when this is set to 1?

Reply:
1 is the default PHP setting

Reply:
Yes, but it looks like we always set `use_cookies` to `0` :thinking:

Reply:
We do so **after** having called `session_start()` the first time

Reply:
ha! It means calling `session_start` adds HTTP headers each time we call it. It could be great to have this information directly as a comment ;)